src/sdk/pynni/nni/compression/torch/pruning/amc/amc_pruner.py

```diff
@@ -0,0 +1,325 @@
+    (https://arxiv.org/pdf/1802.03494.pdf)
+
+    Parameters:
+        model: nn.Module
+            The model to be pruned.
+        config_list: list
+            Configuration list to configure layer pruning.
+            Supported keys:
+                op_types: operation type to be pruned
+                op_names: operation name to be pruned
+        evaluator: function
+            function to evaluate the pruned model.
+            The prototype of the function:
+                >>> def val_func(val_loader, model):
+                >>>     ...
+                >>>     return acc
+        val_loader: torch.utils.data.DataLoader
+            Data loader of validation dataset.
+        suffix: str
+            suffix to help you remember what experiment you ran
+        export: boolean
+            False: search best pruned model and export after search.
+            True: export a searched model, before exporting model, you must run this pruner
+        export_path: str
+            path for exporting models
+
+        # parameters for pruning environment
+        model_type: str
+            model type to prune, currently 'mobilenet' and 'mobilenetv2' are supported.
+        sparsity: float
+            prune ratio
+        lbound: float
+            minimum prune ratio
+        rbound: float
+            maximum prune ratio
+        reward: function
+            reward function type
+
+        # parameters for channel pruning
+        n_calibration_batches: int
+            number of batches to extract layer information
+        n_points_per_layer: int
+            number of feature points per layer
+        channel_round: int
+            round channel to multiple of channel_round
+
+        # parameters for ddpg agent
+        hidden1: int
+            hidden num of first fully connect layer
+        hidden2: int
+            hidden num of second fully connect layer
+        lr_c: float
+            learning rate for critic
+        lr_a: float
+            learning rate for actor
+        warmup: int
+            time without training but only filling the replay memory
+        discount: float
+        bsize: int
+            minibatch size
+        rmsize: int
+            memory size for each layer
+        window_length: int
+        tau: float
+            moving average for target network
+        # noise
+        init_delta: float
+            initial variance of truncated normal distribution
+        delta_decay: float
+            delta decay during exploration
+
+        # parameters for training ddpg agent
+        max_episode_length: int
+            maximum episode length
+        output: str
+            output dir
+        debug: boolean
+            debug mode
+        train_episode: int
+            train iters each timestep
+        epsilon: int
+            linear decay of exploration policy
+        seed: int
+            random seed to set
```

Comment:
for this docstring, it is not clear which one is optional, and what are their default values

Reply:
default values added.